packages/pulumi-aws/src/apps/api/ApiFileManager.ts

```diff
@@ -0,0 +1,174 @@
+        config: {
+            role: role.output.arn,
+            runtime: "nodejs14.x",
+            handler: "handler.handler",
+            timeout: 30,
+            memorySize: 512,
+            description: "Triggered when a file is deleted.",
+            code: new pulumi.asset.AssetArchive({
+                ".": new pulumi.asset.FileArchive(
+                    path.join(app.ctx.appDir, "code/fileManager/manage/build")
+                )
+            }),
+            environment: {
+                variables: { S3_BUCKET: params.fileManagerBucketId }
+            },
+            vpcConfig: params.vpc
+                ? {
+                      subnetIds: params.vpc.subnets.private.map(subNet => subNet.output.id),
+                      securityGroupIds: [params.vpc.vpc.output.defaultSecurityGroupId]
+                  }
+                : undefined
+        }
+    });
+
+    const download = app.addResource(aws.lambda.Function, {
+        name: "fm-download",
+        config: {
+            role: role.output.arn,
+            runtime: "nodejs14.x",
+            handler: "handler.handler",
+            timeout: 30,
+            memorySize: 512,
+            description: "Serves previously uploaded files.",
+            code: new pulumi.asset.AssetArchive({
+                ".": new pulumi.asset.FileArchive(
+                    path.join(app.ctx.appDir, "code/fileManager/download/build")
+                )
+            }),
+            environment: {
+                variables: {
+                    S3_BUCKET: params.fileManagerBucketId,
+                    IMAGE_TRANSFORMER_FUNCTION: transform.output.arn
+                }
+            },
+            vpcConfig: params.vpc
+                ? {
+                      subnetIds: params.vpc.subnets.private.map(subNet => subNet.output.id),
+                      securityGroupIds: [params.vpc.vpc.output.defaultSecurityGroupId]
+                  }
+                : undefined
+        }
+    });
+
+    const manageS3LambdaPermission = app.addResource(aws.lambda.Permission, {
+        name: "fm-manage-s3-lambda-permission",
+        config: {
+            action: "lambda:InvokeFunction",
+            function: manage.output.arn,
+            principal: "s3.amazonaws.com",
+            sourceArn: pulumi.interpolate`arn:aws:s3:::${params.fileManagerBucketId}`
+        },
+        opts: {
+            dependsOn: [/* TODO this.bucket, */ manage.output]
+        }
+    });
+
+    const bucketNotification = app.addResource(aws.s3.BucketNotification, {
+        name: "bucketNotification",
+        config: {
+            bucket: params.fileManagerBucketId,
+            lambdaFunctions: [
+                {
+                    lambdaFunctionArn: manage.output.arn,
+                    events: ["s3:ObjectRemoved:*"]
+                }
+            ]
+        },
+        opts: {
+            dependsOn: [/* TODO this.bucket, */ manage.output, manageS3LambdaPermission.output]
```

Comment:
Should we finalize these TODOs here?

Reply:
The file manager bucket is no longer inside API, so it cannot be passed as a dependency.
Is it safe to remove the dependency?

Reply:
Spoke to @Pavel910 about this. We can remove the TODO.